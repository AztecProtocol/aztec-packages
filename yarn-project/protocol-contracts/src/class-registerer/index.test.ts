import omit from 'lodash.omit';

import { ClassRegistererAddress, getCanonicalClassRegisterer } from './index.js';

describe('ClassRegisterer', () => {
  it('returns canonical protocol contract', () => {
    const contract = getCanonicalClassRegisterer();
    contract.contractClass.privateFunctions.sort((a, b) => a.selector.value - b.selector.value);
    contract.contractClass.publicFunctions.sort((a, b) => a.selector.value - b.selector.value);
    expect(omit(contract, 'artifact')).toMatchSnapshot();
    expect(contract.address.toString()).toEqual(ClassRegistererAddress.toString());
  });
});
