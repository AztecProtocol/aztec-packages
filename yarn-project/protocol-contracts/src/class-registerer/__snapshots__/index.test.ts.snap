// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassRegisterer returns canonical protocol contract 1`] = `
{
  "address": AztecAddress {
    "asBigInt": 12165572618278205561135319266893715614363974000709547171863995507972204751528n,
    "asBuffer": {
      "data": [
        26,
        229,
        120,
        87,
        210,
        210,
        52,
        118,
        43,
        74,
        102,
        9,
        134,
        217,
        77,
        250,
        54,
        27,
        6,
        163,
        9,
        77,
        124,
        119,
        221,
        13,
        19,
        48,
        120,
        81,
        106,
        168,
      ],
      "type": "Buffer",
    },
  },
  "contractClass": {
    "artifactHash": Fr {
      "asBigInt": 1364372310007692800585626333523730889923819928339145276751574119320046781646n,
      "asBuffer": {
        "data": [
          3,
          4,
          53,
          21,
          164,
          29,
          233,
          4,
          193,
          147,
          128,
          246,
          235,
          216,
          242,
          66,
          55,
          97,
          55,
          80,
          84,
          212,
          24,
          60,
          24,
          13,
          127,
          108,
          77,
          239,
          196,
          206,
        ],
        "type": "Buffer",
      },
    },
    "id": Fr {
      "asBigInt": 15823458103336768291861774407076060748007261259174556009564711604871953507490n,
      "asBuffer": {
        "data": [
          34,
          251,
          194,
          137,
          56,
          109,
          23,
          89,
          23,
          38,
          147,
          215,
          101,
          203,
          138,
          228,
          237,
          145,
          243,
          61,
          16,
          54,
          71,
          253,
          142,
          222,
          138,
          190,
          21,
          41,
          228,
          162,
        ],
        "type": "Buffer",
      },
    },
    "packedBytecode": {
      "data": [
        0,
        0,
        0,
        0,
      ],
      "type": "Buffer",
    },
    "privateFunctions": [
      {
        "isInternal": false,
        "selector": FunctionSelector {
          "value": 588837637,
        },
        "vkHash": Fr {
          "asBigInt": 0n,
          "asBuffer": {
            "data": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "type": "Buffer",
          },
        },
      },
      {
        "isInternal": false,
        "selector": FunctionSelector {
          "value": 595568763,
        },
        "vkHash": Fr {
          "asBigInt": 0n,
          "asBuffer": {
            "data": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "type": "Buffer",
          },
        },
      },
      {
        "isInternal": false,
        "selector": FunctionSelector {
          "value": 1669488881,
        },
        "vkHash": Fr {
          "asBigInt": 0n,
          "asBuffer": {
            "data": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "type": "Buffer",
          },
        },
      },
      {
        "isInternal": false,
        "selector": FunctionSelector {
          "value": 2432309179,
        },
        "vkHash": Fr {
          "asBigInt": 0n,
          "asBuffer": {
            "data": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "type": "Buffer",
          },
        },
      },
    ],
    "privateFunctionsRoot": Fr {
      "asBigInt": 14149643440615160691253002398502794418486578915412752764304101313202744681431n,
      "asBuffer": {
        "data": [
          31,
          72,
          106,
          20,
          204,
          180,
          255,
          137,
          103,
          217,
          28,
          197,
          42,
          211,
          3,
          8,
          107,
          19,
          4,
          29,
          137,
          6,
          131,
          49,
          212,
          129,
          225,
          242,
          112,
          231,
          99,
          215,
        ],
        "type": "Buffer",
      },
    },
    "publicBytecodeCommitment": Fr {
      "asBigInt": 13424778679590722659186070167781649503057405647123863047387123265455936508181n,
      "asBuffer": {
        "data": [
          29,
          174,
          39,
          204,
          127,
          226,
          175,
          52,
          95,
          22,
          2,
          83,
          190,
          56,
          117,
          212,
          73,
          167,
          233,
          186,
          107,
          214,
          135,
          71,
          216,
          125,
          78,
          112,
          84,
          184,
          17,
          21,
        ],
        "type": "Buffer",
      },
    },
    "publicFunctions": [],
    "version": 1,
  },
  "instance": {
    "address": AztecAddress {
      "asBigInt": 12165572618278205561135319266893715614363974000709547171863995507972204751528n,
      "asBuffer": {
        "data": [
          26,
          229,
          120,
          87,
          210,
          210,
          52,
          118,
          43,
          74,
          102,
          9,
          134,
          217,
          77,
          250,
          54,
          27,
          6,
          163,
          9,
          77,
          124,
          119,
          221,
          13,
          19,
          48,
          120,
          81,
          106,
          168,
        ],
        "type": "Buffer",
      },
    },
    "contractClassId": Fr {
      "asBigInt": 15823458103336768291861774407076060748007261259174556009564711604871953507490n,
      "asBuffer": {
        "data": [
          34,
          251,
          194,
          137,
          56,
          109,
          23,
          89,
          23,
          38,
          147,
          215,
          101,
          203,
          138,
          228,
          237,
          145,
          243,
          61,
          16,
          54,
          71,
          253,
          142,
          222,
          138,
          190,
          21,
          41,
          228,
          162,
        ],
        "type": "Buffer",
      },
    },
    "initializationHash": Fr {
      "asBigInt": 5411687428194306823799197943510367074755080310726877398642044502900791103391n,
      "asBuffer": {
        "data": [
          11,
          246,
          232,
          18,
          241,
          75,
          176,
          41,
          247,
          203,
          156,
          141,
          168,
          54,
          125,
          217,
          124,
          6,
          142,
          120,
          141,
          79,
          33,
          0,
          127,
          217,
          112,
          20,
          235,
          168,
          207,
          159,
        ],
        "type": "Buffer",
      },
    },
    "portalContractAddress": EthAddress {
      "buffer": {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "type": "Buffer",
      },
    },
    "publicKeysHash": Fr {
      "asBigInt": 17954372136461894568820199868962143715279932924957662285438325406238856880109n,
      "asBuffer": {
        "data": [
          39,
          177,
          208,
          131,
          154,
          91,
          35,
          186,
          241,
          42,
          141,
          25,
          91,
          24,
          172,
          40,
          143,
          207,
          64,
          26,
          251,
          47,
          112,
          184,
          164,
          181,
          41,
          237,
          229,
          250,
          159,
          237,
        ],
        "type": "Buffer",
      },
    },
    "salt": Fr {
      "asBigInt": 1n,
      "asBuffer": {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          1,
        ],
        "type": "Buffer",
      },
    },
    "version": 1,
  },
}
`;
